ledger/src/blockstore.rs

```diff
@@ -4004,32 +4004,14 @@ pub fn create_new_ledger(
     // Explicitly close the blockstore before we create the archived genesis file
     drop(blockstore);
 
-    let archive_path = ledger_path.join(DEFAULT_GENESIS_ARCHIVE);
-    let args = vec![
-        "jcfhS",
-        archive_path.to_str().unwrap(),
-        "-C",
-        ledger_path.to_str().unwrap(),
-        DEFAULT_GENESIS_FILE,
-        blockstore_dir,
-    ];
-    let output = std::process::Command::new("tar")
-        .args(args)
-        .output()
-        .unwrap();
-    if !output.status.success() {
-        use std::str::from_utf8;
-        error!("tar stdout: {}", from_utf8(&output.stdout).unwrap_or("?"));
-        error!("tar stderr: {}", from_utf8(&output.stderr).unwrap_or("?"));
-
-        return Err(BlockstoreError::Io(IoError::new(
-            ErrorKind::Other,
-            format!(
-                "Error trying to generate snapshot archive: {}",
-                output.status
-            ),
-        )));
-    }
+    let archive_path = ledger_path.join(ZSTD_GENESIS_ARCHIVE);
+    let archive_stream =
+        zstd::Encoder::auto_finish(zstd::Encoder::new(File::create(&archive_path)?, 0)?);
+    let mut archive = tar::Builder::new(archive_stream);
+    archive.append_path_with_name(ledger_path.join(DEFAULT_GENESIS_FILE), DEFAULT_GENESIS_FILE)?;
+    archive.append_dir_all("rocksdb", ledger_path.join(blockstore_dir))?;
```

Comment:
I'm not entirely sure I can ask questions here, but anyway, why is `"rocksdb"` here?

Reply:
Building on Richard's answer, the comment for this function mentions yielding a genesis + blockstore (rocksdb) with a slot 0 full of ticks:
https://github.com/solana-labs/solana/blob/ea98c0ea360a89970511301f3ac8e1fe4fd4b537/ledger/src/blockstore.rs#L4066-L4069

Funny enough, embedding `rocksdb` directory into `genesis.tar.bz2` wasn't such a great idea and caused some unintended issues, see https://github.com/solana-labs/solana/issues/23666. As such, we ended up adjusting unpack to ignore the `rocksdb` directory when extracting a genesis archive via https://github.com/solana-labs/solana/pull/25575.

We could/should likely remove the inclusion of the `rocksdb` directory here, but that is beyond the scope of this PR so for now, we should keep the blockstore directory in there

Reply:
Maybe I'm missing something, but it seems to me that in the pre-PR version, the rocksdb database folder is added to the archive under its own name, which is found in `blockstore_dir` (this can be `rocksdb` or `rocksdb_fifo`). Therefore, it seems to me it should be something like this
```
-    archive.append_dir_all("rocksdb", ledger_path.join(blockstore_dir))?;
+    archive.append_dir_all(blockstore_dir, ledger_path.join(blockstore_dir))?;
```

Reply:
I'm inclined to agree with you diman-io, as this function can create level or fifo:
https://github.com/solana-labs/solana/blob/d1d4c1c65478bb85f7773df05da9129d248e792c/ledger/src/blockstore.rs#L4090

So @ripatel-fd, yes, please make the change to use `blockstore` dir (which will be one of `rocksdb` or `rocksdb_fifo`) over hard-coded `rocksdb`. But again, per my earlier point, it shouldn't really matter since these files are ignored on extract anyways:
https://github.com/solana-labs/solana/blob/d1d4c1c65478bb85f7773df05da9129d248e792c/accounts-db/src/hardened_unpack.rs#L523-L528

I'll tear this out and see what (if anything) breaks in CI outside of this PR